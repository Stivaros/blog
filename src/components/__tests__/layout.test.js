import React from "react"
import { render, screen } from "@testing-library/react"
import Layout from "../layout"

describe("Layout", () => {
  describe("when on root path", () => {
    it("renders title and all posts link", () => {
      const location = {
        pathname: '/'
      }

      render(<Layout location={location} title="Test Title" />)

      expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument()
      expect(screen.getByRole('link', { name: 'Test Title | All Posts' })).toBeInTheDocument()
      expect(screen.getByText(/Built with/)).toBeInTheDocument()
      expect(screen.getByRole('link', { name: 'Gatsby' })).toBeInTheDocument()
    })
  })

  describe("when not on root path", () => {
    it("renders back link", () => {
      const location = {
        pathname: '/blog-post'
      }

      render(<Layout location={location} title="Test Title" />)

      expect(screen.getByRole('heading', { level: 3 })).toBeInTheDocument()
      expect(screen.getByRole('link', { name: /Back to Test Title Index/i })).toBeInTheDocument()
    })
  })
})
