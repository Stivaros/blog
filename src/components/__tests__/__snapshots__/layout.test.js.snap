// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout when on root path renders correctly 1`] = `
<div
  style={
    Object {
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": "34.8rem",
      "padding": "2.175rem 1.0875rem",
    }
  }
>
  <header>
    <h1
      style={
        Object {
          "marginBottom": "2.175rem",
          "marginTop": 0,
        }
      }
    >
      <a
        href="/"
      >
         | All Posts
      </a>
    </h1>
  </header>
  <main />
  <footer>
    Â© Efstathios Stivaros 
    2023
    , Built with
     
    <a
      href="https://www.gatsbyjs.org"
    >
      Gatsby
    </a>
  </footer>
</div>
`;
